src/runtime/contrib/tensorrt/tensorrt_module.cc

```diff
@@ -61,6 +60,9 @@ class TensorRTModule : public runtime::ModuleNode {
     for (auto& it : trt_engine_cache_) {
       it.second.context->destroy();
       it.second.engine->destroy();
+      for (void* ptr : it.second.device_mem_pointers) {
+        TVMDeviceFreeDataSpace({kDLGPU, 0}, ptr);
```

Comment:
Can you check to make sure the ptr isn't null first? Since sometimes not every input will have a buffer and may still be nullptr

Reply:
good catch, I was thinking it, didn't end up making it into the code. I will send an update.